### PR TITLE
Feature gdpr api

### PR DESCRIPTION
**Pull request by AntiAki-M**
## Description :sparkles:
First version of GDPR API implementation
Allows fast access to general details on users documents and whether they are deletable or under contractual obligation

### Closes :no_good_woman:
**[ATV-126](https://helsinkisolutionoffice.atlassian.net/browse/ATV-126): Add deletable field to document model** 
**[ATV-127](https://helsinkisolutionoffice.atlassian.net/browse/ATV-127): Implement GDPR feature** 
### Automated tests :gear:️
Yes

### Manual testing :construction_worker_man:
Yes
## Additional notes :spiral_notepad:
API is only usable with API key authentication and requires users uuid as a lookup field

### CODE REVIEW

File: documents/api/viewsets.py [hunk 1]
```diff
@@ -37,22 +37,81 @@
...
+    queryset = Document.objects.none()
+    serializer_class = GDPRSerializer
+    pagination_class = None
+    filter_backends = [DjangoFilterBackend]
+    lookup_field = "user__uuid"
+    http_method_names = ["get", "delete"]
+
+    def get_queryset(self):
+        user = self.request.user
+        if user.is_anonymous:
```
Comment (vinikvuo) [inline documents/api/viewsets.py:69]: I suppose there are permission classes for this

File: documents/api/viewsets.py [hunk 2]
```diff
@@ -37,22 +37,81 @@
...
+    def destroy(self, request, *args, **kwargs):
+        with self.record_action():
+            # Anonymize data
+            with transaction.atomic():
+                qs = self.filter_queryset(self.get_queryset()).filter(
+                    deletable=True, **kwargs
+                )
+                Attachment.objects.filter(document__in=qs).delete()
+                qs.update(user=None, content={}, business_id="")
+            return self.retrieve(request, **kwargs)
```
Comment (vinikvuo) [inline documents/api/viewsets.py:96]: Why retrieve is called?

File: documents/serializers/attachment.py [hunk 3]
```diff
@@ -8,6 +8,18 @@
...
 
 
+class GDPRAttachmentSerializer(serializers.ModelSerializer):
+    """Basic "read" serializer for the Attachment model"""
+
+    class Meta:
+        model = Attachment
+        fields = ("filename",)
+
+    def to_representation(self, instance):
```
Comment (vinikvuo) [inline documents/serializers/attachment.py:18]: What this do compared to default to_representation

Comment (vinikvuo) [reply documents/serializers/attachment.py:18]: I suppose this has something to do with DocumentSerializer

File: documents/api/viewsets.py (hunk 2 above)
Comment (AntiAki-M) [reply documents/api/viewsets.py:96]: Retrieve is called to return user the documents that were left after deletion. This was decided with the project architect.

File: documents/serializers/attachment.py (hunk 3 above)
Comment (AntiAki-M) [reply documents/serializers/attachment.py:18]: This is used for formating the gdpr response. Once again decided with the architect. It's certainly a bit odd, but necessary for the response we needed.

File: documents/api/viewsets.py (hunk 1 above)
Comment (AntiAki-M) [reply documents/api/viewsets.py:69]: This follows the code style of the project.

File: documents/api/viewsets.py (hunk 2 above)
Comment (vinikvuo) [reply documents/api/viewsets.py:96]: I'm pretty sure that REST API should not work like this.

**Pull request merged**